server.js

```diff
@@ -1,12 +1,41 @@
 const express = require('express');
 const app = express();
+const cors = require("cors");
 
-app.get('/getme', (req, res) => {
-  res.send(fruits)
+app.use(express.json({extended: false}))
+app.use(cors())
+
+
+app.get('/getme', (req, res, next) => {
+  res.send({message: `${req.url} path`})
 });
 
-app.post('/postme', (req, res) => {
-  res.send({})
+app.get("/fuckup/saw-it-coming",(req,res,next)=>{
+  try{
+    res.sen("hello")
+  }catch(err){
+    next(err)
+  }
+})
+
+app.get("/fuckup/surprise",(req,res)=>{
+  throw new Error("There is a dummy error!")
+
+})
+
+app.post('/postme', (req, res, next) => {
+  res.json({message: "POST route called"})
+})
+
+
+// 404 MIDDLEWARE
+app.use((req,res,next)=>{
+    res.status(404).json({error: `Path ${req.url} does not exist...`})
+})
+
+// GENERIC ERROR MIDDLEWARE
+app.use((err,req,res,next)=>{
+  res.status(404).json({error: err.message})
```

Comment:
Almost all perfect until here :-)

Currently you send back hardcoded error status 404 in your generic error handler. 

But we wanna be flexible, meaning, every route should be able to set the adequate response code when calling our error handler.

The default error code instead of 404 should be 500 (generic error case) . But that code could be overwritten by a more explicit error code, e.g. 404 in case of unknown route.

So e.g. if somebody requested in ther user/:id route a user with an ID that does not exist you typically set a property "code" or "status" with 404 on your error and then forward it to your generic error handler. 

Sending correctly 404 and 500 is quite important, as the frontend typically will handle them differently...

404 indicates to the frontend developer, he probably coded in something wrong in his fetch. He called the backend incorrectly.
500 indicates to the frontend developer, the backend is to blame. He can now show that error to the backend dev and force him to fix it :)